pkg/promutil/migrate.go

```diff
@@ -101,40 +101,53 @@ func BuildMetrics(results []model.CloudwatchMetricResult, labelsSnakeCase bool,
 			}
 
 			for _, statistic := range statisticsInCloudwatchData(metric) {
-				dataPoint, ts, err := getDatapoint(metric, statistic)
-				if err != nil {
-					return nil, nil, err
+				datapoints, err := getDatapoints(metric, statistic)
+				for _, dataPoint := range datapoints {
+					ts := dataPoint.Timestamp
+					dataPoint := dataPoint.Datapoint
+					if err != nil {
+						return nil, nil, err
+					}
+					var exportedDatapoint float64
+					if dataPoint == nil && metric.MetricMigrationParams.AddCloudwatchTimestamp {
+						// If we did not get a datapoint then the timestamp is a default value making it unusable in the
+						// exported metric. Attempting to put a fake timestamp on the metric will likely conflict with
+						// future CloudWatch timestamps which are always in the past. It's safer to skip here than guess
+						continue
+					}
```

Comment:
I think there's a slight change in behavior here, if the feature is disabled and the first datapoint is nil we will bypass the break at the end potentially exporting an old data point as current. Could you add a test for this use case to see if it happens?

Reply:
Your right, there must be a different behaviour here depending if we want to export historic data. Thanks for pointing that out!